### PR TITLE
Port is_subset to`TreeMap`

### PR DESCRIPTION
**Pull request by ariesdevil**
Port `is_subset` to `TreeMap`

### CODE REVIEW

File: croaring/src/treemap/imp.rs [hunk 1]
```diff
@@ -652,4 +652,58 @@ impl Treemap {
...
+           match other.map.get(k) {
+               None => return false,
+               Some(other_v) => {
+                   if !v.is_subset(other_v) {
+                       return false;
+                   }
+               }
+           }
+       }
+        true
```
Comment (Dr-Emann) [inline croaring/src/treemap/imp.rs:686]: The indentation is a little off here

**Review by Dr-Emann — COMMENTED** [review]
Small nit about indentation, otherwise looks good to me

File: croaring/src/treemap/imp.rs [hunk 2]
```diff
@@ -652,4 +652,58 @@ impl Treemap {
...
+            match other.map.get(k) {
+                None => return false,
+                Some(other_v) => {
+                    if !v.is_subset(other_v) {
+                        return false;
+                    }
+                }
+            }
+        }
+        true
```
Comment (Dr-Emann) [inline croaring/src/treemap/imp.rs:686]: Depends on how you feel, noting that this _could_ be written more shortly as

```rust
self.map.iter().all(|(key, lhs)| {
    lhs.is_empty() || other.map.get(key).map_or(false, |rhs| lhs.is_subset(rhs))
})
```

**Review by Dr-Emann — APPROVED** [review]
Looks good to me. I added a shorter version, but I'd be fine with what you have.

File: croaring/src/treemap/imp.rs (hunk 2 above)
Comment (ariesdevil) [reply croaring/src/treemap/imp.rs:686]: So if it's accepted, let's merge it?

**Comment by saulius** [comment]
Thanks for the contribution @ariesdevil! This will be part of 1.0 containing [some future-proof, yet breaking changes](https://github.com/RoaringBitmap/croaring-rs/issues/108).

**Pull request merged**